requestmatch/matcher.go

```diff
@@ -114,6 +115,24 @@ func compileRxs(exps []string) ([]*regexp.Regexp, error) {
 	return rxs, nil
 }
 
+func keyValToLower(kv map[string]string) map[string]string {
```

Comment:
this and the one below is actually a painful example how generics could be good sometimes